core/src/main/scala/org/apache/spark/Aggregator.scala

```diff
@@ -54,14 +55,13 @@ case class Aggregator[K, V, C] (
       }
       combiners.iterator
     } else {
-      val combiners = new ExternalAppendOnlyMap[K, V, C](createCombiner, mergeValue, mergeCombiners)
-      combiners.insertAll(iter)
-      // Update task metrics if context is not null
       // TODO: Make context non optional in a future release
-      Option(context).foreach { c =>
-        c.taskMetrics.memoryBytesSpilled += combiners.memoryBytesSpilled
-        c.taskMetrics.diskBytesSpilled += combiners.diskBytesSpilled
-      }
+      val spillMetrics = Option(context).map(
+        _.taskMetrics.getOrCreateShuffleReadSpillMetrics()).getOrElse(new WriteMetrics())
```

Comment:
Ah yeah, looks like this is a copy/paste mistake.  In this path, I think we're always on the map side, so the call should just be `getOrCreateShuffleWriteSpillMetrics`.